### PR TITLE
Refs #5148 - Add api for host package applicability

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

File: db/migrate/20160520175340_add_host_applicable_package.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class AddHostApplicablePackage < ActiveRecord::Migration
+  def change
+    create_table "katello_content_facet_applicable_rpms" do |t|
+      t.references 'content_facet', :null => false
+      t.references 'rpm', :null => false
+    end
```
Comment (daviddavis) [inline db/migrate/20160520175340_add_host_applicable_package.rb]: Should the combination of content_facet -> rpm be unique? If so, what about adding a unique index to this migration?

**Comment by daviddavis** [comment]
Code looks good. Going to test it out as I work on the UI/CLI.

File: db/migrate/20160520175340_add_host_applicable_package.rb (hunk 1 above)
Comment (jlsherrill) [inline db/migrate/20160520175340_add_host_applicable_package.rb]: added index

File: app/controllers/katello/api/v2/packages_controller.rb [hunk 2]
```diff
@@ -13,6 +14,50 @@ def auto_complete_name
...
+    param :host_id, :number, :desc => N_("host id to list applicable packages for")
+    param :packages_restrict_installable, :boolean, :desc => N_("Only show packages that are installable")
+    param :packages_restrict_applicable, :boolean, :desc => N_("Only show packages that are applicable to hosts (defaults to true if host_id is specified)")
+    param_group :search, ::Katello::Api::V2::ApiController
+    def index
+      super
+    end
+
+    def custom_index_relation(collection)
+      if params[:packages_restrict_installable]
```
Comment (daviddavis) [inline app/controllers/katello/api/v2/packages_controller.rb]: I'm testing this out with my UI code and it looks like when the UI passes in 'false', it's a string (which evals to true). You probably need to cast it like we do for errata:

https://git.io/vi2UH

Comment (jlsherrill) [inline app/controllers/katello/api/v2/packages_controller.rb]: fixed

**Comment by jlsherrill** [comment]
updated api to use packages_restrict_upgradable

**Comment by daviddavis** [comment]
ACK

**Pull request merged**